packages/renderer/src/modules/server.ts

```diff
@@ -0,0 +1,85 @@
+import { MessageTransport, RPC } from '@dcl/mini-rpc';
+
+import { fs } from '#preload';
+import { type Project } from '/shared/types/projects';
+
+export enum Method {
+  READ_FILE = 'read_file',
+  WRITE_FILE = 'write_file',
+  EXISTS = 'exists',
+  DELETE = 'delete',
+  LIST = 'list',
+}
+
+export type Params = {
+  [Method.READ_FILE]: {
+    path: string;
+  };
+  [Method.WRITE_FILE]: {
+    path: string;
+    content: Buffer;
+  };
+  [Method.DELETE]: {
+    path: string;
+  };
+  [Method.EXISTS]: {
+    path: string;
+  };
+  [Method.LIST]: {
+    path: string;
+  };
+};
+
+export type Result = {
+  [Method.READ_FILE]: Buffer;
+  [Method.WRITE_FILE]: void;
+  [Method.DELETE]: void;
+  [Method.EXISTS]: boolean;
+  [Method.LIST]: {
+    name: string;
+    isDirectory: boolean;
+  }[];
+};
+
+export type Callbacks = { [key in Method]: (params: Params[key]) => Result[key] };
```

Comment:
yeaaah, it's actually well typed since we the param/return type are infering with the same key.

![Jul-29-2024 11-40-07](https://github.com/user-attachments/assets/3e940cc5-0152-4497-971d-20c594873bd1)


but anyway, I agree with the comment below, so I will replace this 🤣